src/main/java/com/mercadolibrewannabe/controller/ProductController.java

```diff
@@ -0,0 +1,46 @@
+package com.mercadolibrewannabe.controller;
+
+import com.mercadolibrewannabe.model.Product;
+import com.mercadolibrewannabe.model.User;
+import com.mercadolibrewannabe.model.form.ProductForm;
+import com.mercadolibrewannabe.repository.CategoryRepository;
+import com.mercadolibrewannabe.repository.ProductRepository;
+import com.mercadolibrewannabe.utils.Uploader;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/product")
+public class ProductController {
+
+	private final ProductRepository productRepository;
+	private final CategoryRepository categoryRepository;
+
+	private final Uploader uploader;
+
+
+	public ProductController (ProductRepository productRepository,
+	                          CategoryRepository categoryRepository,
+	                          Uploader uploader) {
+
+		this.productRepository = productRepository;
+		this.categoryRepository = categoryRepository;
+		this.uploader = uploader;
+	}
+
+	@PostMapping
+	@ResponseStatus(code = HttpStatus.CREATED)
+	public void create(@Valid ProductForm productForm) {
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Aqui você pode usar a annotation @AuthenticatedPrincipal ou algo assim. Exemplo de código:

```
(@Valid ProductForm... , @AuthenticatedPrincipal User user)
```

Reply:
Massa! Não conhecia. Sabia o fato de poder usar o @CreatedBy na classe de domain